Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -162,14 +129,25 @@ public actor KafkaProducer {
             }
         }
 
+        // Kill poll loop in polling system
+        self.pollingSystem.terminate()
```

Comment:
I am not sure that this is enough. We should figure out what happens in a system where we don't poll anymore but still produce e.g. nobody consumes the acknowledgements sequence and it gets back pressured. Will rdkafka buffer then and at some point also back pressure or will we just OOM?

Reply:
I believe [`rd_kafka_flush()`](https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#aaff06c4372bce917c17f3c1a5d8b205d) got us covered here (we already use this).